sdk/core/allocator/main.cc

```diff
@@ -305,27 +333,8 @@ namespace
 
 					revoker.system_bg_revoker_kick();
 
-					if constexpr (Revocation::Revoker::IsAsynchronous)
-					{
-						wait_for_background_revoker(
-						  timeout, needsRevocation->waitingEpoch, g);
-					}
-					else
-					{
-						// Drop and reacquire the lock while yielding.
-						// Sleep for a single tick.
-						g.unlock();
-						Timeout smallSleep{1};
-						if (thread_sleep(&smallSleep) < 0)
-						{
-							/* Unable to sleep; bail out */
-							return nullptr;
-						}
-						if (!reacquire_lock(timeout, g, smallSleep.elapsed))
-						{
-							return nullptr;
-						}
-					}
+					wait_for_background_revoker(
```

Comment:
Should this check the return value?

Reply:
Yes.  That was an existing bug, but it's now fixed.